src/chains/ethereum/ethereum/src/api.ts

```diff
@@ -83,6 +86,114 @@ async function autofillDefaultTransactionValues(
   }
 }
 
+function createSimulatedTransaction(
+  blockchain: Blockchain,
+  options: EthereumInternalOptions,
+  common: Common,
+  transaction: Ethereum.Transaction,
+  simulationBlock: Block
+): SimulationTransaction {
+  const simulationBlockHeader = simulationBlock.header;
+
+  let gas: Quantity;
+  if (typeof transaction.gasLimit === "undefined") {
+    if (typeof transaction.gas !== "undefined") {
+      gas = Quantity.from(transaction.gas);
+    } else {
+      // eth_call isn't subject to regular transaction gas limits by default
+      gas = options.miner.callGasLimit;
+    }
+  } else {
+    gas = Quantity.from(transaction.gasLimit);
+  }
```

Comment:
Just a tangentially related though - this "getGasFromTransaction" feels like something that we do in a couple different places, might be worth trying to centralise it at some stage 🤷 (but probs not here and now)

Reply:
(and maybe the same for "getInputDataFromTransaction" as per below.